Code/Framework/AzFramework/AzFramework/Network/IRemoteTools.h

```diff
@@ -0,0 +1,243 @@
+        bool m_isBlobOwner;
+        bool m_immediateSelfDispatch;
+    };
+
+    using RemoteToolsMessagePointer = AZStd::intrusive_ptr<RemoteToolsMessage>;
+    using RemoteToolsMessageQueue = AZStd::deque<RemoteToolsMessagePointer, AZ::OSStdAllocator>;
+    
+    // id for the local application
+    static constexpr AZ::u32 s_selfNetworkId = 0xFFFFFFFF;
+
+    class RemoteToolsEndpointInfo final
+    {
+
+    public:
+        AZ_CLASS_ALLOCATOR(RemoteToolsEndpointInfo, AZ::OSAllocator, 0);
+        AZ_TYPE_INFO(RemoteToolsEndpointInfo, "{DD0E9B2A-3B25-43B1-951E-CACCEC5D6754}");
+
+        explicit RemoteToolsEndpointInfo(AZStd::string displayName = AZStd::string{}, AZ::u32 networkId = 0)
+            : m_displayName(AZStd::move(displayName))
+            , m_persistentId(0)
+            , m_networkId(networkId)
+        {
+        }
+
+        //! Gets if this endpoint is ourself
+        //! @return true if this endpoint is ourself
+        bool IsSelf() const;
+
+        //! Gets if this endpoint is online
+        //! @return true if this endpoint is online
+        bool IsOnline() const;
+
+        //! Gets if this endpoint is valid
+        //! @return true if this endpoint is named, has a network ID and persistent ID
+        bool IsValid() const;
+
+        //! Gets the display name of this endpoint
+        //! @return A string display name of this endpoint
+        const char* GetDisplayName() const;
+
+        //! Gets the Persistent ID of this endpoint
+        //! @return A Crc32 representing the Persistent ID of this endpoint
+        AZ::u32 GetPersistentId() const;
+
+        //! Gets the AzNetworking Network ID of this endpoint
+        //! @return The AzNetworking Network ID for the connection of this endpoint
+        AZ::u32 GetNetworkId() const;
+
+        //! Sets information used to determine validity of the endpoint
+        //! @param displayName The string display name of this endpoint
+        //! @param persistentId The Crc32 based persistent ID of this endpoint
+        //! @param networkId The AzNetworking Network ID for this endpoint's connection
+        void SetInfo(AZStd::string displayName, AZ::u32 persistentId, AZ::u32 networkId);
+
+        bool IsIdentityEqualTo(const RemoteToolsEndpointInfo& other) const;
+
+    private:
+        AZStd::string m_displayName;
+        AZ::u32 m_persistentId; // this is a CRC key used to identify a RemoteTools target
+        AZ::u32 m_networkId; // this is the connection id, used for AzNetworking communications.
+        bool m_isOnline;
+    };
+
+    using RemoteToolsEndpointContainer = AZStd::unordered_map<AZ::u32, RemoteToolsEndpointInfo>;
+    using RemoteToolsEndpointStatusEvent = AZ::Event<RemoteToolsEndpointInfo>;
+    using RemoteToolsEndpointConnectedEvent = AZ::Event<bool>;
+    using RemoteToolsEndpointChangedEvent = AZ::Event<AZ::u32, AZ::u32>;
+    using ReceivedRemoteToolsMessages = AZStd::fixed_vector<RemoteToolsMessagePointer, 64>;
```

Comment:
Well it's fixed so the size has to be declared up front. 64 specifically because it was the size used originally in TargetManagement and I took that to mean it's an appropriate size for routine usage.